sonar-db/src/main/java/org/sonar/db/issue/IssueDao.java

```diff
@@ -47,27 +56,59 @@ public void selectNonClosedIssuesByModule(long componentId, ResultHandler handle
     }
   }
 
-  @CheckForNull
-  public IssueDto selectNullableByKey(DbSession session, String key) {
-    return mapper(session).selectByKey(key);
+  public Optional<IssueDto> selectByKey(DbSession session, String key) {
+    return Optional.fromNullable(mapper(session).selectByKey(key));
   }
 
-  public IssueDto selectByKey(DbSession session, String key) {
-    IssueDto issue = selectNullableByKey(session, key);
-    if (issue == null) {
-      throw new IllegalArgumentException(String.format("Issue key '%s' does not exist", key));
+  public IssueDto selectByKeyOrFail(DbSession session, String key) {
+    Optional<IssueDto> issue = selectByKey(session, key);
+    if (!issue.isPresent()) {
+      throw new RowNotFoundException(String.format("Issue with key [%s] does not exist", key));
     }
-    return issue;
+    return issue.get();
   }
 
-  public List<IssueDto> findByActionPlan(DbSession session, String actionPlan) {
+  public List<IssueDto> selectByActionPlan(DbSession session, String actionPlan) {
     return mapper(session).selectByActionPlan(actionPlan);
   }
 
+  /**
+   * Gets a list issues by their keys. The result does NOT contain {@code null} values for issues not found, so
```

Comment:
Do you think this description is useful ? Isn't it the default behavior ? I wouldn't say the same thing for selectByOrderedKeys